### PR TITLE
Initial pass at Device model

### PR DESCRIPTION
**Pull request by danielmorrison**
This is my thinking about how a `Device` model could work. 

You'd specify an array or `usernames` and `projects` that you care about, for example: 

```ruby
Device.new(
  name: "Office Light",
  identifier: "some-external-id",
  usernames: ["collectiveidea", "deadmanssnitch"],
  projects: ["inchworm/inchworm", "client/foo"]
)
```

Then there's a simple `Status#devices` method to find the ones that care about a status. The idea being we'd send a ping to each of them. (I was going to do an `OR` query to do it in one instead of 2, but I'll wait until Rails 5 when that's built in :trollface:)

I'm using postgres arrays, so it requires us to switch to postgres for dev/test.

What else do we need?

### CODE REVIEW

File: db/migrate/20160510201736_create_devices.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateDevices < ActiveRecord::Migration
+  def change
+    enable_extension 'uuid-ossp'
+
+    create_table :devices, id: :uuid do |t|
+      t.string :usernames, array: true, default: [], null: false
+      t.string :projects, array: true, default: [], null: false
+
+      t.timestamps null: false
```
Comment (albus522) [inline db/migrate/20160510201736_create_devices.rb]: Needs a `external_device_id`

Comment (albus522) [inline db/migrate/20160510201736_create_devices.rb]: Possibly a `service` but only one supported for now

Comment (danielmorrison) [inline db/migrate/20160510201736_create_devices.rb]: cool. I was wondering, and the PR got my question answered! 😄

Comment (danielmorrison) [inline db/migrate/20160510201736_create_devices.rb]: I called it `identifier`. No need for `service` until we have a 2nd one.

**Comment by tbugai** [comment]
needs to be ♻️'d

**Comment by danielmorrison** [comment]
@tbugai ♻️ complete.

**Pull request merged**